React/React Pong/src/components/Board.tsx

```diff
@@ -0,0 +1,187 @@
+                    setGameStarted(true);
+                    break;
+            }
+        }
+
+        window.addEventListener("keydown", handleKeyDown);
+
+        return () => {
+            window.removeEventListener("keydown", handleKeyDown);
+        };
+    }, [leftPaddle, rightPaddle]);
+
+    function moveBall() {
+        setBall({ x: ball.x + velocity.x, y: ball.y + velocity.y });
+    }
+
+    function getPaddlePosition(paddle: number) {
+        return [paddle, paddle + 1, paddle + 2, paddle + 3, paddle + 4];
+    }
+
+    function checkPaddleCollision() {
+        const leftPaddlePosition = getPaddlePosition(leftPaddle);
+        const rightPaddlePosition = getPaddlePosition(rightPaddle);
+
+        if (ball.y + 1 == 30) {
+            if (rightPaddlePosition.some((part) => ball.x == part)) {
+                const newVelocity =
+                    Math.random() > 0.5 ? { x: -1, y: -1 } : { x: 1, y: -1 };
+                setVelocity(newVelocity);
+                setDirection(DIRECTIONS.LEFT);
+                setWallHit(false);
+            } else {
+                setWallHit(true);
+            }
+        }
+
+        if (ball.y - 1 == 0) {
+            if (leftPaddlePosition.some((part) => ball.x == part)) {
+                const newVelocity =
+                    Math.random() > 0.5 ? { x: 1, y: 1 } : { x: -1, y: 1 };
+                setVelocity(newVelocity);
+                setDirection(DIRECTIONS.RIGHT);
+                setWallHit(false);
+            } else {
+                setWallHit(true);
+            }
+        }
+    }
+
+    function checkBorderCollision() {
+        if (ball.x == 1) {
+            if (direction == DIRECTIONS.RIGHT) {
+                setVelocity({ x: 1, y: 1 });
+            } else {
+                setVelocity({ x: 1, y: -1 });
+            }
+        } else if (ball.x == 19) {
+            if (direction == DIRECTIONS.RIGHT) {
+                setVelocity({ x: -1, y: 1 });
+            } else {
+                setVelocity({ x: -1, y: -1 });
+            }
+        }
+    }
+
+    function checkCollision() {
+        checkBorderCollision();
+        checkPaddleCollision();
+    }
+
+    function ballSpeed() {
+        if (speed > 40) {
+            setSpeed(speed - 2);
+        }
+    }
+
+    function newBall() {
+        const randomDir =
+            Math.random() > 0.5 ? { x: 1, y: 1 } : { x: 1, y: -1 };
+        if (randomDir.y == -1) {
+            setDirection(DIRECTIONS.LEFT);
+        } else {
+            setDirection(DIRECTIONS.RIGHT);
+        }
+        setBall(initialBallState);
+        setVelocity(randomDir);
+        ballSpeed();
+    }
+
+    function scorePoint() {
+        if (ball.y == 0) {
+            setPlayerTwoScore(playerTwoScore + 1);
+            newBall();
+        } else if (ball.y == 30) {
+            setPlayerOneScore(playerOneScore + 1);
+            newBall();
+        }
+    }
+
+    useEffect(() => {
+        const interval = setInterval(() => {
```

Comment:
Move the interval in a useRef()
Set  the interval only if it has not been set already.